File: python/examples/cells/cells.py
"""Application logic for cells.uimd - extends generated UI."""

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))

from runtime import UIApplication
from cells_ui import CellsUI


class Cells(CellsUI):
    """Custom cells with application logic."""

    def open(self):
        super().open()

    def save(self):
        pass

    def focused(self, element, value):
        pass


def main():
    app = UIApplication()
    window = Cells()
    app.open(window)
    return app.run()


if __name__ == "__main__":
    sys.exit(main())

File: rust/examples/cells/cells.rs
mod cells_ui;

use cells_ui::*;

#[derive(Default)]
struct CellsApp;

impl CellsUIEvents for CellsApp {}

fn main()
{
    let code = uimd::run_generated_app_main(||
    {
        let mut ui = CellsUI::new();
        let mut app = CellsApp;
        ui.run(&mut app, &std::env::args().collect::<Vec<_>>())
    });
    std::process::exit(code);
}
